.eslintrc.json

```diff
@@ -24,6 +24,7 @@
     "@typescript-eslint/ban-ts-comment": "off",
     "@typescript-eslint/no-explicit-any": "off",
     "@typescript-eslint/no-inferrable-types": "off",
-    "import/no-cycle": "error"
+    "import/no-cycle": "error",
+    "@typescript-eslint/no-var-requires": "off"
```

Comment:
Long story w/this one, but essentially I added this because: 
- Our `eslint` config is set up to expect `es6` modules (uses `import` for imports) on the line `"sourceType": "module" `. 
- Our `tsconfig` file, though, tells the compiler to use CJS syntax (uses `require` for imports)
- So, linting complains about the `require` statements in our `utils/` files without the addition of this rule to our `eslint` config 

We should make whatever syntax we want to use consistent throughout the repo to avoid confusing users at some point, but for now this is the most minimal change that allows me to bypass linting issues.

Reply:
Deeper dive:

Q: Why doesn't my IDE complain about the files in dirs such as `src/` using `import` instead of `require`, if `tsconfig` says we're supposed to be writing CJS?
- This is because we also have the line "esModuleInterop": true` in our `tsconfig` file. This allows us to use ESM syntax (`import`) in our Typescript files, even though the underlying module system is CJS. 

Q: Why don't you just change the imports in the `utils/` files to use `import` (ESM) instead of `require` (CJS) then?
- Wwhen I tried [this](https://github.com/pinecone-io/pinecone-ts-client/actions/runs/10636403435/job/29489510906), I ran into the following error: 
<img width="877" alt="Screenshot 2024-08-30 at 11 40 55 AM" src="https://github.com/user-attachments/assets/ebeb56cc-6ca8-43e0-811c-7e90326f0a6d">
- I think this we run the files in `utils/` with the `node` command, and `node` does not understand `typescript` or `es6` (ESM) module syntax. By default, `node` expects `CJS` syntax.

Reply:
Interesting, thanks for doing a deep dive on this. I see the IDE issues as well, but since we specifically lint `src/` I think that's why this isn't generally an issue in CI and whatever.

I'd personally prefer something more specific more specific than disabling for the whole repo. I think we should probably look at cleaning some of these utility files up a bit.

Reply:
[ignore this -- made a ticket to address in a separate PR: https://app.asana.com/0/1203260648987893/1208186698556902/f]